### ISSUE TITLE
[BUG] Wrong path for artifacts when using http server

### ISSUE THREAD

**Issue by Roffild** [mlflow/mlflow#2671]
When the server starts, the path is set via the command or created ./mlruns/
The mlflow.get_artifact_uri() function should receive the absolute path from the server, if it is a local path. Or return the FTP path, if specified in the settings.
```python
import mlflow
mlflow.set_tracking_uri("http://127.0.0.1:5000/")
mlflow.set_experiment("AutoPytorch")
with mlflow.start_run(run_name="runtest"):
    print(mlflow.get_artifact_uri()) # ./mlruns/1/fbad648257e549b0843bcb996e91a175/artifacts
```
When using mlflow.log_artifact(), the files are saved in the script folder, not the server.

The function mlflow.set_artifact_uri() is missing.

**Comment by mronda** [comment]
They way I found to go about this is to specify your artifact location when starting the server. I set the same directory for both store-uri and artifact-root. 
```
mlflow server --backend-store-uri file:///sample_directory/mlflow --default-artifact-root file:///sample_directory/mlflow -h 0.0.0.0
```

**Comment by joglekara** [comment]
> They way I found to go about this is to specify your artifact location when starting the server. I set the same directory for both store-uri and artifact-root.
> 
> ```
> mlflow server --backend-store-uri file:///sample_directory/mlflow --default-artifact-root file:///sample_directory/mlflow -h 0.0.0.0
> ```

You're right, this does work when the directories are the same. However, there are times when it is useful to have the information split into two different directories (different types of storage, SSD vs. HDD etc.). 

So just want to echo what @Roffild said. It would add to an already software to have this working.

**Comment by harupy** [comment]
@Roffild Did you set `artifact_location` when you created `AutoPytorch` experiment?

**Comment by nickvazz** [comment]
Any updates?

**Comment by HansBambel** [comment]
> @Roffild Did you set `artifact_location` when you created `AutoPytorch` experiment? If you didn't, artifacts would be saved in local.

I am not sure whether this is fully related, but I start a server locally with:
`mlflow server --backend-store-uri sqlite:///mlflow.db --default-artifact-root artifacts --host 0.0.0.0` in my project root. Then when I run a training script in `scripts/train.py` where I set
```
mlflow.set_tracking_uri("http://localhost:5000")
mlflow.set_experiment("my_experiment")
```
The artifacts get saved to `scripts/artifacts` instead of the root `artifacts`-folder. Therefore, they do not show up in the UI (only when I copy them manually from the `scripts/artifacts` to `artifacts`).

I tried setting and not setting an artifact folder when creating a new experiment as you suggested. Not setting one results in the UI displaying the artifact-folder as `./artifacts/1` and setting it results in the UI displaying it as `my_exp`. But both still get saved under `scripts` instead of the root.

How can I save the artifacts and models under the correct folder?

**Comment by afparsons** [comment]
Perhaps relatedly, I am also having difficulty sending artifacts to an S3 bucket from within a Docker container.

**Comment by farrandi** [comment]
Having the same issue.
Specified my server in a remote desktop:
```
mlflow server --backend-store-uri postgresql://mlflow_user@hostname/mlflow_db --default-artifact-root ftp://mlflow@hostname/mlruns --host 0.0.0.0 --port 5000
```

And in my client (different desktop) I have the script (in `.py` form):
```
REMOTE_SERVER_URI = "http://hostname:5000/"
mlflow.set_tracking_uri(REMOTE_SERVER_URI)
print(mlflow.tracking.get_tracking_uri())   #http://hostname:5000/
mlflow.start_run()
print(mlflow.get_artifact_uri())   #file:./mlruns/0/76fd760e2e1c4a9ea6114f5445094430/artifacts
```
The artifact uri is the local `file:./mlruns` instead of the `ftp:/` server. I have made sure my ftp server works and connects from my client.

**Comment by farrandi** [comment]
Got mine to work!
I had to edit a table value in my `--backend-store-uri` in my case it was postgres, so I changed the `artifact_location` shown here 
![image](https://user-images.githubusercontent.com/54917098/143954400-d3f22acc-d29c-4e45-9253-797fb511443b.png)
the values by default was: `file:./mlruns/[id]`

**Comment by gabrielhazeredo** [comment]
> > @Roffild Did you set `artifact_location` when you created `AutoPytorch` experiment? If you didn't, artifacts would be saved in local.
> 
> I am not sure whether this is fully related, but I start a server locally with: `mlflow server --backend-store-uri sqlite:///mlflow.db --default-artifact-root artifacts --host 0.0.0.0` in my project root. Then when I run a training script in `scripts/train.py` where I set
> 
> ```
> mlflow.set_tracking_uri("http://localhost:5000")
> mlflow.set_experiment("my_experiment")
> ```
> 
> The artifacts get saved to `scripts/artifacts` instead of the root `artifacts`-folder. Therefore, they do not show up in the UI (only when I copy them manually from the `scripts/artifacts` to `artifacts`).
> 
> I tried setting and not setting an artifact folder when creating a new experiment as you suggested. Not setting one results in the UI displaying the artifact-folder as `./artifacts/1` and setting it results in the UI displaying it as `my_exp`. But both still get saved under `scripts` instead of the root.
> 
> How can I save the artifacts and models under the correct folder?

The only thing that worked for me was manually assigning to the experiment

mlflow.create_experiment("my_experiment", "../artifacts")
mlflow.set_tracking_uri("http://localhost:5000")
mlflow.set_experiment("my_experiment")

**Comment by in-oleynikov** [comment]
Hello!
Have similiar problem with the path. I use s3-storage in local minio to store artifacts and postgres to store params and results
Server starting like this:
```
mlflow server --backend-store-uri postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@db/${POSTGRES_DB} --default-artifact-root s3://artifacts/ --host 0.0.0.0
``` 
But when I'm asking for artifacts uri:
```
mlflow.get_artifact_uri()
```
it returns string:
```
 s://artifacts/0/c902a63de7ce467da0f9dfe9aab8f6f8/artifacts
```
So it returns s:// instead of s3://
And as expected error:
```
mlflow.exceptions.MlflowException: Could not find a registered artifact repository for: s://artifacts/0/c902a63de7ce467da0f9dfe9aab8f6f8/artifacts. Currently registered schemes are: ['', 'file', 's3', 'gs', 'wasbs', 'ftp', 'sftp', 'dbfs', 'hdfs', 'viewfs', 'runs', 'models', 'http', 'https', 'mlflow-artifacts']
```
How to solve such issue?

**Comment by in-oleynikov** [comment]
Solved with just this line
```
mlflow.set_experiment("AutoPytorch")
```

**Comment by jorgelmadrid** [comment]
I solved it using An HTTP URI like
``` 
http://my-tracking-server:5000
```
